### PR TITLE
osx: thread starts before the assignments

### PR DESCRIPTION
**Pull request by obastemur**
Changed the order of the member assignments since the thread may start just before the parameters are assigned. 

This especially happens when the osx scheduler is very busy.

### CODE REVIEW

File: src/unix/stream.c [hunk 1]
```diff
@@ -353,16 +353,20 @@ int uv__stream_try_select(uv_stream_t* stream, int* fd) {
   s->fake_fd = fds[0];
   s->int_fd = fds[1];
 
-  if (uv_thread_create(&s->thread, uv__stream_osx_select, stream))
-    goto fatal4;
-
   s->stream = stream;
   stream->select = s;
+  const int old_fd = *fd;
```
Comment (indutny) [inline src/unix/stream.c]: C89 style declarations should be at the top of the block.

**Comment by indutny** [comment]
One small nit, otherwise LGTM. Thanks for catching it!

**Comment by obastemur** [comment]
Done. Thanks.

**Comment by indutny** [comment]
Thank you, landed in a6ff04d2c48be408cdcd76e5d78f4dc0a8fb9312

**Pull request closed without merging**